lib/CloudinaryPlugin.js

```diff
@@ -98,20 +97,16 @@ class CloudinaryPlugin {
       }
     ];
 
-    this.config = givenConfig;
+    this.config = config;
 
     const { NODE_ENV } = process.env;
-    const missingCredMessage = 'Cloudinary Plugin cannot be used without the ApiKey, the ApiSecret and the CloudName of your Cloudinary Account';
-
     const { cloudinaryCloudName, cloudinaryApiKey, cloudinaryApiSecret } = this.config;
 
-    //Check if credentials are missing 
     if (!cloudinaryApiKey || !cloudinaryApiSecret || !cloudinaryCloudName) {
       if (NODE_ENV === 'production') {
```

Comment:
For that kind of feature, I firmly believe that this test should be reversed: we allow to continue _only_ if NODE_ENV explicitly set to development.
That way, the plugin will crash if NODE_ENV is not set at all, and it seems way safer to me.

```suggestion
      if (NODE_ENV !== 'development') {
```